### PR TITLE
Simplify few loops by using memcpy

### PR DESCRIPTION
**Pull request by mmaret**
Simplify code and may produce better performances

### CODE REVIEW

**Comment by AJenbo** [comment]
Thanks, had considered this but wasn't sure how it should be applied :)

File: Source/engine.cpp [hunk 1]
```diff
@@ -291,31 +270,10 @@ void CelBlitLightSafe(BYTE *pDecodeTo, BYTE *pRLEBytes, int nDataSize, int nWidt
 			if (!(width & 0x80)) {
 				i -= width;
 				if (dst < gpBufEnd && dst > gpBufStart) {
-					if (width & 1) {
-						dst[0] = tbl[src[0]];
```
Comment (AJenbo) [inline Source/engine.cpp]: This broke object rendering, tbl[] is the lighting table so everything was fully lit.

**Comment by AJenbo** [comment]
Fixed https://github.com/diasurgical/devilutionX/commit/1df595c1e0bdd77add85025b70be64914a781b58

**Comment by mmaret** [comment]
Sorry for that... May be I should avoid commiting while moving the baby
swing with my feet :)
I was sure to have it tested before sending the MR...

On another side, how do you measure time rendering performance on your side
@ajenbo ?

Thanks !

Le mar. 22 oct. 2019 à 01:40, Anders Jenbo <notifications@github.com> a
écrit :

**Comment by AJenbo** [comment]
No worries, I should have done a better job of testing before merging as well. 

Put a timer that measures the time different between the start and end of DrawAndBlit()

**Comment by mmaret** [comment]
This have indeed, on my system, increase the performances.
DrawAndBlit get from ~23ms to 21ms

**Comment by AJenbo** [comment]
Awsome that is about half of the performance lost by the cleanup, so we are already on a surprisingly good track with optimizing things again :)

What kind of system is this by the way? On my Raspberry Pi 3B I get 13ms. Or maybe this was a debug build and not a release build? It would be best to measure things on release builds as they are both better optimized and do not have all the debug hooks.

**Pull request merged**